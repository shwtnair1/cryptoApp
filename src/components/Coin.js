import React from 'react';

const Coin = (props)=>{
    
    return (
        <div className='coin-container'>
            <div>
                <img src={props.image} alt='crypto' />
                <p id='coin-name'>{props.name}</p>
            </div>
            <div>
                <p id='coin-price'>Rs.{props.price}</p>
            </div>
            <div>
            {props.pricechange<0 ? (
                        <p id='coin-pricechange' className="red">{props.pricechange.toFixed(2)}%</p>
                    ):(
                        <p id='coin-pricechange' className="green">{props.pricechange.toFixed(2)}%</p>
                    )}
            </div>
            <div>
                <p id='coin-cap'>
                    Mkt Cap: Rs.{props.marketcap.toLocaleString()}
                </p>
            </div>
        </div>
    )
}

export default Coin;