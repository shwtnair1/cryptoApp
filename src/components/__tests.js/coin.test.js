import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Coin from '../Coin';
Enzyme.configure({ adapter: new Adapter() });

const coin = {
    "image":"https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
    "name":"Bitcoin",
    "price":3106272,
    "pricechange":3.44971,
    "marketcap":1990031721822
}
describe('Coin',()=>{
    const instance = shallow(<Coin {...coin}/>);

    it('should have props',()=>{
        expect(instance.props()).toBeDefined()
    })

    it('should check snapshot',()=>{
        expect(instance).toMatchSnapshot()
    })
    
    it('should display coin symbol image',()=>{
        const imageSrc = instance.find('div div img').prop('src');
        const imageAlt = instance.find('div div img').prop('alt');
         expect(imageSrc).toBe('https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579');
         expect(imageAlt).toBe('crypto')
    })

    it('should display coin symbol name',()=>{
        const nameElement = instance.find('#coin-name');
         expect(nameElement.text()).toBe('Bitcoin');
    })

    it('should display coin symbol price',()=>{
        const priceElement = instance.find('#coin-price');
         expect(priceElement.text()).toBe('Rs.3106272');
    })

    it('should display green coin price change',()=>{
        const priceElement = instance.find('#coin-pricechange');
         expect(priceElement.text()).toBe('3.45%');
         expect(priceElement.prop('className')).toBe('green');
    })

    it('should display coin market cap',()=>{
        const capElement = instance.find('#coin-cap');
         expect(capElement.text()).toBe('Mkt Cap: Rs.19,90,03,17,21,822');
    })
})