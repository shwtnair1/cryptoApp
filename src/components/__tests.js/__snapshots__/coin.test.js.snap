// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin should check snapshot 1`] = `ShallowWrapper {}`;
